transifex/utils.py

```diff
@@ -0,0 +1,342 @@
+                    'commit', '-m', self.message
+                ],
+                check=True
+            )
+
+    def push(self):
+        """Push branch to the remote."""
+        subprocess.run(['git', 'push', '-u', 'origin', self.branch_name], check=True)
+
+    def open_pr(self):
+        """Create a new PR on GitHub."""
+        if self.pr:
+            raise RuntimeError(
+                'A pull request has already been opened. [{repo}/{pr}]'.format(
+                    repo=self.name,
+                    pr=self.pr.number
+                )
+            )
+
+        self.pr = self.github_repo.create_pull(
+            self.message,
+            'This PR was created by a script.',
+            'master',
+            self.branch_name
+        )
+
+    def merge_pr(self):
+        if not self.pr:
+            raise RuntimeError(
+                'A pull request has not been opened for {repo}:{branch}.'.format(
+                    repo=self.name,
+                    branch=self.branch_name
+                )
+            )
+
+        if self.pr.is_merged():
+            raise RuntimeError(
+                'The pull request [{repo}/{pr}] has already been merged.'.format(
+                    repo=self.name,
+                    pr=self.pr.number
+                )
+            )
+
+        retries = 0
+        while retries <= self.MAX_MERGE_RETRIES:
+            try:
+                self.pr.merge(merge_method=self.merge_method)
+                logger.info('Merged [%s/#%d].', self.name, self.pr.number)
+                return True
+            except github.GithubException as e:
+                # Assumes only one commit is present on the PR.
+                statuses = self.pr.get_commits()[0].get_statuses()
+
+                # Check for any failing Travis builds. If any are found, notify the team
+                # and move on.
+                if any('travis' in s.context and s.state == 'failure' for s in statuses):
+                    logger.info(
+                        'A failing Travis build prevents [%s/#%d] from being merged. Notifying %s.',
+                        self.name, self.pr.number, self.owner
+                    )
+
+                    self.pr.create_issue_comment(
+                        '@{owner} a failed Travis build prevented this PR from being automatically merged.'.format(
+                            owner=self.owner
+                        )
+                    )
+
+                    return False
+                else:
+                    logger.info(
+                        'Status checks on [%s/#%d] are pending. This is retry [%d] of [%d].',
+                        self.name, self.pr.number, retries, self.MAX_MERGE_RETRIES
+                    )
+
+                    retries += 1
+
+                    if retries <= self.MAX_MERGE_RETRIES:
+                        # Poll every 5 minutes
+                        time.sleep(60 * 5)
+        else:
+            logger.info(
```

Comment:
I would prefer not to address this here, as this is one of several issues I noticed with the existing script that I decided not to do anything about during this pass of refactoring. I have created issues on the repository for each of the problems I identified that I would like to see fixed going forward. See https://github.com/edx/ecommerce-scripts/issues